services/ui-src/src/App.js

```diff
@@ -53,14 +56,16 @@ function App() {
           <Navbar.Collapse>
             <Nav pullRight>
               {isAuthenticated ? (
-                <>
-                  <NavDropdown id="User" title={email}>
-                    <LinkContainer to="/profile">
-                      <NavItem>User Profile</NavItem>
-                    </LinkContainer>
-                    <NavItem onClick={handleLogout}>Logout</NavItem>
-                  </NavDropdown>
-                </>
+                setEmailFromUserInfo() && (
```

Comment:
It's a little strange to me to be calling this function inside render. Like, what happens if currentUserInfo() takes a long time? Would it fix it to make the condition `{ isAuthenticated && email ? ...`

That way you only render the logout button after currentUesrInfo has happened.

Reply:
Ahh ok, I gave that a try... 
```
    <Nav pullRight>
              {isAuthenticated && email ? (
```
The behavior I see is broken but in a different way.  Instead of the email displaying correctly or being blank, only the SignUp/Login buttons appear in the top right.  Screenshot below:
<img width="1339" alt="Screen Shot 2021-04-12 at 1 42 46 PM" src="https://user-images.githubusercontent.com/48921055/114438238-7d53ac00-9b95-11eb-8dbb-7b001f4abe0c.png">
I think isAuthenticated is getting set from the Login route, but nothing in the login workflow - after logging in - is triggering the email to found and set.

Reply:
I'm not real confident in all this here...

Reply:
As far as I can tell the onLoad() function should be called by your effect, which should end up calling setEmail. I'd try and debug why that's not getting called here. 

React-wise, I think it makes the most sense for that call to currentUserInfo() to be in an effect, and then your render function will be purely dependent on your state, rather than including a promise function call in your render function.

Reply:
If that doesn't get you anywhere let me know and I can ask Hana to take a look.

Reply:
Yeah something doesn't seem right here.
Thanks I'll step through it some more and try to figure out what's going on (or not going on).

Reply:
I wonder if you need to await the onLoad() call in useEffect

Reply:
```
./src/App.js
  Line 18:5:  Parsing error: Can not use keyword 'await' outside an async function

  16 | 
  17 |   useEffect(() => {
> 18 |     await onLoad();
     |     ^
  19 |   }, []);
  20 | 
  21 |   async function onLoad() {
```
:/

Reply:
make the useEffect function async as well: 

`useEffect(async () => {`

Reply:
eh, here's someone making it seems like the way it is should be fine: https://github.com/facebook/react/issues/14326

Reply:
Ahhhh ok, that compiled fine.  That produces the original "blank on first login.... correct after refresh and subsequent navigation" behavior.

Reply:
Ok so I did a little more debugging...
I'm running a version without these changes, with the broken behavior...
I modified useEffect() to the following:
```
seEffect(() => {
    onLoad();
    console.log("INSIDE USEEFFECT, AFTER ONLOAD");
  }, []);
  ```
  Stepping through the steps to reproduce...
  when i go to the site for the frist time incognito, i see the log statement in the logs.
  I log in...
  I don't see that log statement made a second time.  So I guess my take is, the heart of the matter is that useEffect (which calls onLoad) is NOT getting called when we are sent from /login back to /.  And I guess to firm up my questions some, they'd be:
  - Is that expected behavior
  - In either case... why?

Reply:
Ahhh, ok  well that makes some sense. The effect is set to only run on component mount and unmount with the [] second argument. So if you load the page not-logged-in, onLoad() is not going to run after you do login, you need some way to trigger it. 

You could pass a callback into the login component, or figure out how to get a prop passed in. 

We have an authContext that we use to manage the auth state that then lower components can useAuth() to get props from it.

Reply:
This has been really helpful, thank you thank you.  
So this sounds like this bad behavior is to be expected.
This workaround seems like the wrong approach.  I'm not certain on this but it feels wrong.  I'm going to mark this as a draft and see if I can get a clearer picture in at least my head.

Reply:
@macrael Things are a little more clear, or at least more familiar.  The current / -> /login -> / workflow does not trigger onEffect().  And that's why the current (prior to these changes) location of the setEmail() function, inside onLoad, will not work as intended.  
I've remembered that this functionality (the profile section and the email label) was something I added to the repo.  So the call to setEmail and its location is entirely on me.  Which makes me even more suspicious this isn't set up right.
- I'm of the mind to remove the calls to setEmail entirely, and just use a 'My Profile' label in the header, which was something I had written as an option into the issue.  This stuff works, but if it's making an antipattern more of an antipattern, I'd rather just bail on the "show my email" functionality.  
- You mentioned Hana might be able to weigh in and shed some light on this; that'd be most welcome and appreciated.  If some insight can save this issue from the "My Profile" option that'd be awesome, but any clue into best practices and good patterns is appreciated.